File: atcoder.jp/contests/abc143/tasks/abc143_c/main.hs
dedup :: Eq a => [a] -> [a]
dedup [] = []
dedup [x] = [x]
dedup (x : y : xs) =
  if x == y
    then dedup (y : xs)
    else x : dedup (y : xs)

main :: IO ()
main = do
  n <- readLn :: IO Int
  s <- getLine

  let ans = length . dedup $ s

  print ans

File: atcoder.jp/contests/abc143/tasks/abc143_c/main.rs
use proconio::input;

fn main() {
    input! {
        _n: i32,
        s: String,
    }

    let ans = s.dedup().len();

    println!("{}", ans);
}

pub trait MyString {
    fn dedup(&self) -> String;
}

impl MyString for str {
    fn dedup(&self) -> String {
        let mut vec = self.as_bytes().to_vec();
        vec.dedup();
        String::from_utf8(vec).unwrap()
    }
}
